DirectXMesh/DirectXMeshOptimizeLRU.cpp

```diff
@@ -508,9 +520,11 @@ HRESULT DirectX::OptimizeFacesLRU(
         return E_INVALIDARG;
 
     if ((uint64_t(nFaces) * 3) >= UINT32_MAX)
-        return HRESULT_FROM_WIN32(ERROR_ARITHMETIC_OVERFLOW);
+        return HRESULT_E_ARITHMETIC_OVERFLOW;
 
+#ifdef WIN32
     InitOnceExecuteOnce(&s_initOnce, ComputeVertexScores, nullptr, nullptr);
```

Comment:
You could use `std::call_once` (C++11) as a cross-platform replacement for this instead of using a function-local static, if you wanted. You could also just use the function-local static for all platforms, I suspect.

Reply:
The function-local static relies on 'magic statics' which can be turned off for Windows so I kept the Win32 API...